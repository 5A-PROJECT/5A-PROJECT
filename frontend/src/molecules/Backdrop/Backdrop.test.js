import React from 'react';
import { render } from '@testing-library/react';
import BackDrop from '.';

const TestComponent = (loading) => render(<BackDrop loading={loading} />);

describe('<Backdrop />', () => {
  it('스냅샷과 일치해야 합니다.', () => {
    const { container } = TestComponent(true);
    expect(container).toMatchSnapshot();
  });

  it('loading = true이면 렌더링 합니다.', () => {
    const { getByTestId } = TestComponent(true);
    expect(getByTestId('backdrop')).toBeTruthy();
  });

  it('loading = false 이면 렌더링 하지 않습니다.', () => {
    const { queryByTestId } = TestComponent(false);
    expect(queryByTestId('backdrop')).toBeNull();
  });
});
