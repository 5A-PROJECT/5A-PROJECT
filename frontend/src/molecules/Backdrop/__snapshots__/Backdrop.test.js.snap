// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Backdrop /> 스냅샷과 일치해야 합니다. 1`] = `
<div>
  <div
    class="sc-AxiKw cXiqAS"
    data-testid="backdrop"
  >
    <span
      class="sc-AxhCb iFQXpm"
    >
      <div
        class="sc-AxjAm kSkoow"
      >
        <div
          class="MuiCircularProgress-root sc-AxirZ ktnnZK MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
      </div>
    </span>
  </div>
</div>
`;
